rest/config_manager.go

```diff
@@ -761,56 +761,56 @@ func (b *bootstrapContext) ComputeMetadataIDForDbConfig(ctx context.Context, con
 //  4. The _default._default collection isn't already associated with a different metadata ID (syncInfo document is not present, or has a value of defaultMetadataID)
 func (b *bootstrapContext) computeMetadataID(ctx context.Context, registry *GatewayRegistry, config *DbConfig) string {
 
-	standardMetadataID := b.standardMetadataID(config.Name)
-
-	// If there's already a metadataID assigned to this database in the registry (including other config groups), use that
-	defaultMetadataIDInUse := false
-	for _, cg := range registry.ConfigGroups {
-		for dbName, db := range cg.Databases {
-			if dbName == config.Name {
-				return db.MetadataID
-			}
-			if db.MetadataID == defaultMetadataID {
-				defaultMetadataIDInUse = true
+	metadataID, source := func() (string, string) {
```

Comment:
We generally should be using base.MD() redaction for dbnames (and should probably for metadataID as well).  What about changing this function to return `metadataID, dbName, sourceMsg`, and then change the log line (812) to something like
```
base.InfofCtx(ctx, "Using metadata ID %v for db %v, reason: %v, base.MD(metadataID), base.MD(dbName), msg)
```
We would lose the ability to log additional metadataID or dbNames with this approach, which we do in two places, though.   In general this makes me wonder whether the anonymous function is really the right approach, or whether we should just do the logging everywhere we return and be diligent in future when changing this function.